sparse/impl/KokkosSparse_sptrsv_cuSPARSE_impl.hpp

```diff
@@ -436,6 +436,198 @@ void sptrsvcuSPARSE_solve(KernelHandle* sptrsv_handle,
 #endif
 }
 
+// --------------------------------
+// Stream interface
+// --------------------------------
+
+template <class ExecutionSpace, class KernelHandle,
+          class ain_row_index_view_type, class ain_nonzero_index_view_type,
+          class ain_values_scalar_view_type, class b_values_scalar_view_type,
+          class x_values_scalar_view_type>
+void sptrsvcuSPARSE_solve_streams(
+    const std::vector<ExecutionSpace> &execspace_v,
+    std::vector<KernelHandle> &handle_v,
+    const std::vector<ain_row_index_view_type> &row_map_v,
+    const std::vector<ain_nonzero_index_view_type> &entries_v,
+    const std::vector<ain_values_scalar_view_type> &values_v,
+    const std::vector<b_values_scalar_view_type> &rhs_v,
+    std::vector<x_values_scalar_view_type> &lhs_v, bool /*trans*/
+) {
+#ifdef KOKKOSKERNELS_ENABLE_TPL_CUSPARSE
+  using idx_type         = typename KernelHandle::nnz_lno_t;
+  using size_type        = typename KernelHandle::size_type;
+  using scalar_type      = typename KernelHandle::nnz_scalar_t;
+  using memory_space     = typename KernelHandle::HandlePersistentMemorySpace;
+  using sptrsvHandleType = typename KernelHandle::SPTRSVHandleType;
+  using sptrsvCuSparseHandleType =
+      typename sptrsvHandleType::SPTRSVcuSparseHandleType;
+
+  int nstreams = execspace_v.size();
+#if (CUDA_VERSION >= 11030)
```

Comment:
@vqd8a should this be `CUSPARSE_VERSION`?

Reply:
@ndellingwood I have seen there a inconsistent use of CUDA_VERSION and CUSPARSE_VERSION in Kokkos Kernels, so I just picked one. Moreover, there are mismatch between CUDA_VERSION and CUSPARSE_VERSION that we have to be careful. For example, CUDA_VERSION 11030 (11.3.0) has CUSPARSE_VERSION 11500 (cusparse 11.5.0). What should we do here?

Reply:
@vqd8a yeah, the CUDA vs CUSPARSE versioning mismatch is unfortunate (without building/testing code to query the version, one way to check is to view the lib directory of the cuda install, where the cusparse version is appended to the lib)

Just to clarify, you confirmed that the cusparse api you need here is present with the cusparse version in cuda/11.3.0, and the other api (in the `#else` branch) is present in older versions? If so, then I think what you have done is fine

Reply:
@ndellingwood This is correct, I just looked at cuda versions only.